### ISSUE TITLE
manual bridge upgrade process

### ISSUE THREAD

**Issue by adriatic** [monterey-framework/monterey#101]
to be written ...

**Comment by adriatic** [comment]
Saved the discussion on this topic in **[this wiki](https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge/wiki)** as **[How to update aurelia framework for a generic aurelia application](https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge/wiki/How-to-update-aurelia-framework-for-a-generic-aurelia-application)**

**Comment by adriatic** [comment]
Any comments that explain this behavior?

**Comment by JeroenVinke** [comment]
> aurelia-framework updated to 1.0.1 in the "root section of the map" - why does aurelia-binding not appear in that section

aurelia-binding is not a top level dependency of your app, it is a dependency of aurelia-framework, which **is** a top level dependency of your app. I believe that aurelia-binding is not mentioned in package.json as a dependency


![img](https://cloud.githubusercontent.com/assets/2712405/17255626/05101e84-5588-11e6-905f-308538f95728.png)

This means that aurelia-templating has a dependency on aurelia-binding.

**Comment by adriatic** [comment]
I agree - and would like to offer a more complete analysis of this process:

### Step 1 - define `package.json`
Aurelia developer (Rob in this case) defines the list of Aurelia Framework components that the app needs in **[this section of the package.json](https://github.com/aurelia/skeleton-navigation/blob/master/skeleton-esnext/package.json#L68-L82)**. I could only guess that the app developer defined this set by ___knowing___ what this skeleton needs - or by listing more components that the skeleton needs, "just to be safe"

### Step 2 - run `jspm install`
Note that if jspm is invoked by `jspm install` command (with no list of modules to fetch), jspm will check for list of modules in the list defined in **[this section of the package.json](https://github.com/aurelia/skeleton-navigation/blob/master/skeleton-esnext/package.json#L68-L82)** - and in each case will create the **[`config.js`]()** file.

**Comment by JeroenVinke** [comment]
Good to have these things written down. Did you try "some-package@*" in package.json and then to jspm update? (Not sure is is jspm or npm)

**Comment by adriatic** [comment]
Thanks @JeroenVinke - I will certainly try that as well, as we cannot spend too much time on these tests, if we want to do something good with app version management :-)

By the way, check [this](point_up: [July 29, 2016 3:51 PM](https://gitter.im/aurelia-ui-toolkits/team?at=579bb3ce79f75971375200a6)) where I am trying to influence others to take this very seriously

**Comment by adriatic** [comment]
@JeroenVinke give me a better example for 
>Did you try "some-package@*" in package.json

from this list
```
  "jspm": {
    "dependencies": {
      "aurelia-animator-css": "npm:aurelia-animator-css@^1.0.0-rc.1.0.0",
      "aurelia-bootstrapper": "npm:aurelia-bootstrapper@^1.0.0-rc.1.0.0",
      "aurelia-fetch-client": "npm:aurelia-fetch-client@^1.0.0-rc.1.0.0",
      "aurelia-framework": "npm:aurelia-framework@^1.0.0-rc.1.0.0",
      "aurelia-history-browser": "npm:aurelia-history-browser@^1.0.0-rc.1.0.0",
      "aurelia-loader-default": "npm:aurelia-loader-default@^1.0.0-rc.1.0.0",
      "aurelia-logging-console": "npm:aurelia-logging-console@^1.0.0-rc.1.0.0",
      "aurelia-pal-browser": "npm:aurelia-pal-browser@^1.0.0-rc.1.0.0",
      "aurelia-polyfills": "npm:aurelia-polyfills@^1.0.0-rc.1.0.0",
      "aurelia-router": "npm:aurelia-router@^1.0.0-rc.1.0.0",
      "aurelia-templating-binding": "npm:aurelia-templating-binding@^1.0.0-rc.1.0.0",
      "aurelia-templating-resources": "npm:aurelia-templating-resources@^1.0.0-rc.1.0.0",
      "aurelia-templating-router": "npm:aurelia-templating-router@^1.0.0-rc.1.0.0",
      "bluebird": "npm:bluebird@3.4.1",
      "bootstrap": "github:twbs/bootstrap@^3.3.6",
      "fetch": "github:github/fetch@^1.0.0",
      "font-awesome": "npm:font-awesome@4.6.3",
      "jquery": "npm:jquery@^2.2.4",
      "text": "github:systemjs/plugin-text@^0.0.8"
    },
```

**Comment by JeroenVinke** [comment]
First downgrade fontawsome to lets say 4.0.0 (if that version exists) by changing 4.6.3 in package.json into 4.0.0 and run jspm install. Then change `"font-awesome": "npm:font-awesome@4.0.0",` to `"font-awesome": "npm:font-awesome@*",` and jspm install. If jspm doesnt throw an error (which probably means this is an npm feature and not jspm) check what it did

**Comment by JeroenVinke** [comment]
Follow up: what if you remove the version number of font awsome in package json and jspm update

**Comment by Thanood** [comment]
Did you also delete the `jspm_packages` directory in your second test?
If not:

> Removing package files for npm:font-awesome@4.6.3. Where did jspm remove it from??

From the `jspm_packages` directory. Remember jspm packages are stored with their version number in their directory/file pair.

> Is Removing package files for github:systemjs/plugin-css@0.1.25 consequence of 1. above?

Partly yes/no. As `font-awesome` depends on a css loader it is installed as a dependency. It is **not** installed as a top-level dependency (which jspm reads from your `package.json`) so it is removed. 

Install any other jspm package and remove it from `package.json` to confirm. jspm removed it because nothing depends on it. :smile:

**Comment by adriatic** [comment]
___@Thanood - you ask many good questions and make many good points - deserving more of my attention than I can afford at this very moment. I will come back to this in a couple of hours___
